gui: add option to delete wallet

jp1ac4 opened the pull request:
This is to resolve #543.

I've used a trash icon instead of a cross, but can change it to a cross if preferred.

darosior commented:
Played a bit with this. Using a trash icon is fine by me.

The presentation isn't the prettiest but fine, clear and usable:
![image](https://github.com/wizardsardine/liana/assets/22457751/7273beb5-c1d8-4a49-98a7-41a8f8d707af)

The modal however i think could be improved. Currently:
![image](https://github.com/wizardsardine/liana/assets/22457751/7f40cbcf-9702-40c8-9e43-2e369f81fadb)
I tried to align the content to the center (as i think in all other modals?) but i'm not sure it really improves things:
![image](https://github.com/wizardsardine/liana/assets/22457751/6ab0c485-87e5-4abb-b2e9-563ef71e919d)
I'd like to grey-out the line about the managed bitcoind or convey it's less important information. Also i think the WARNING line should be last. Here is my best attempt so far:
![image](https://github.com/wizardsardine/liana/assets/22457751/5b35628c-6c55-4963-aa31-0e1bd702d310)

What do you think? Here is my diff in case you want to take it / improve over:
```diff
diff --git a/gui/src/launcher.rs b/gui/src/launcher.rs
index 8ef01fc..70b3c7c 100644
--- a/gui/src/launcher.rs
+++ b/gui/src/launcher.rs
@@ -276,8 +276,8 @@ impl DeleteWalletModal {
             "Are you sure you want to delete the wallet and all associated data for {}?",
             wallet_name(&self.network)
         );
-        let help_text_2 = "WARNING: This cannot be undone.";
-        let help_text_3 = "If you are using a Liana-managed Bitcoin node, it will not be affected by this action.";
+        let help_text_2 = "(If you are using a Liana-managed Bitcoin node, it will not be affected by this action.)";
+        let help_text_3 = "WARNING: This cannot be undone.";
 
         Into::<Element<ViewMessage>>::into(
             card::simple(
@@ -289,7 +289,8 @@ impl DeleteWalletModal {
                             .width(Length::Fill),
                     ))
                     .push(Row::new().push(text(help_text_1)))
-                    .push(Row::new().push(text(help_text_2)))
+                    .push(Row::new().push(caption(help_text_2).style(color::GREY_3).small()))
+                    .push(Row::new())
                     .push(Row::new().push(text(help_text_3)))
                     .push_maybe(self.warning.as_ref().map(|w| {
                         notification::warning(w.to_string(), w.to_string()).width(Length::Fill)
@@ -302,9 +303,12 @@ impl DeleteWalletModal {
                             .align_items(Alignment::Center)
                             .push(icon::circle_check_icon().style(color::GREEN))
                             .push(text("Wallet successfully deleted").style(color::GREEN))
-                    }),
+                    })
+                    .align_items(Alignment::Center),
             )
-            .width(Length::Fixed(700.0)),
+            .width(Length::Fixed(700.0))
+            .center_x()
+            .center_y(),
         )
         .map(Message::View)
     }
```

darosior commented on gui/src/launcher.rs:
```diff
@@ -164,6 +225,89 @@ pub enum Message {
+    fn update(&mut self, message: Message) -> Command<Message> {
+        if let Message::View(ViewMessage::DeleteWallet(DeleteWalletMessage::Confirm)) = message {
+            self.warning = None;
+            if let Err(e) = std::fs::remove_dir_all(&self.wallet_datadir) {
```
This would remove the bitcoind watchonly wallet file. Is it an issue? Depends. It's certainly not the best to be removing a file managed by a process from under it. On the other hand it's unlikely to make bitcoind crash, and even if it did i don't think it would be worse than any other alternative i can think of for handling this.

Therefore i think it's fine like that, but wanted to share in case other think about something i would have missed.

jp1ac4 replied:
Would it be worth adding a message relating to this in the modal (if they are not using Liana-managed node for that wallet)?

darosior replied:
I don't think so.

darosior reviewed (commented):
Tested ACK aside from my artistic sensibilities.

jp1ac4 commented:
> I tried to align the content to the center (as i think in all other modals?) but i'm not sure it really improves things

Perhaps we could keep left-alignment for the text and then center for the button, which is how the installer modal for editing keys is.

> I'd like to grey-out the line about the managed bitcoind or convey it's less important information.

It makes sense to grey it out. I could also check whether that wallet is using a Liana-managed node and only show the message in that case.

jp1ac4 commented:
One other thing that might be nice would be to see the wallet be deleted in the blurred background while the modal is open. Currently, it's only removed from the list when the user closes the modal.

edouardparis commented on gui/src/launcher.rs:
```diff
@@ -96,31 +137,42 @@ impl Launcher {
-                                                                    }
-                                                                    _ => theme::Badge::Standard,
-                                                                }),
+                                                Row::new()
```
add some spacing

jp1ac4 commented:
I've made changes based on the feedback so far. I've kept them as separate commits for now in case we only want some of them, but I'll then squash them.

In https://github.com/wizardsardine/liana/pull/864/commits/91d4c9a385f2ac7cda03071b859c56de55e74cda:
- added spacing between wallet button and trash icon
- changed wording order and formatting of text (I used `p1_regular` for managed bitcoind message as it should be equivalent to `caption().small()`)
- changed alignment of "Delete wallet" button to centre

In https://github.com/wizardsardine/liana/pull/864/commits/8b8300a2136465f5b9a65ae26615c72b6507782f, we now refresh the list of available wallets in the blurred background while the modal is still open.

In https://github.com/wizardsardine/liana/pull/864/commits/c32f4b9d040eec881e57e253394d2d3175d061d3, I check whether the wallet uses managed bitcoind. There are now three different cases:

1) Wallet uses managed bitcoind:
![image](https://github.com/wizardsardine/liana/assets/121959000/1fc8151e-f997-49a8-970a-11bb5da62311)

2) Wallet does not use managed bitcoind:
![image](https://github.com/wizardsardine/liana/assets/121959000/255827cf-8de1-4246-8a8c-af8029ae4698)

3) We are not sure if wallet uses managed bitcoind (error reading config file):
![image](https://github.com/wizardsardine/liana/assets/121959000/af8cea4a-3d13-4c4c-9ab2-f6d123cd02f1)

darosior commented:
This looks great to me. From some light testing it's really slick. Will let @edouardparis have a look at the code before merging.

Slightly-tested-not-reviewed ACK c32f4b9d040eec881e57e253394d2d3175d061d3

edouardparis commented:
ACK c32f4b9d040eec881e57e253394d2d3175d061d3

jp1ac4 commented:
I'll squash commits now.

edouardparis commented:
ACK 279c1c75a1dfe6617798b59cbe3932c803aeb90f

darosior commented:
Great work.
